File: lib/medium/sort_colors.dart
void sortColors(List<int> nums) {
  Map<int, int> freq = {};
  for (int i in nums) {
    freq[i] = (freq[i] ?? 0) + 1;
  }
  int index = 0;
  for (int i = 0; i < 3; i++) {
    int count = freq[i] ?? 0;
    while (count-- > 0) {
      nums[index++] = i;
    }
  }
}

File: src/main/php/Medium/SortColors.php
<?php

declare(strict_types=1);

namespace Src\Medium;

class SortColors
{
    public function solve(array &$nums)
    {
        $freq = array_count_values($nums);
        $index = 0;
        for ($i = 0; $i < 3; $i++) {
            $count = $freq[$i];
            while ($count-- > 0) {
                $nums[$index++] = $i;
            }
        }
    }
}
